_posts/2018-07-11-cockpit-172.md

```diff
@@ -0,0 +1,77 @@
+---
+title: Cockpit 172
+author: pitti
+date: 2018-07-11
+tags: cockpit linux
+slug: cockpit-172
+category: release
+summary: Cockpit with on-demand PCP install, flow control, and Python 3 support
+comments: 'true'
+---
+
+Cockpit is the [modern Linux admin interface](https://cockpit-project.org/). We release regularly.
+Here are the release notes from version 172.
+
+### System: Offer installation of PCP
+
+The System page now shows an "Enable persistent metrics…" link if the
+[PCP](https://pcp.io/) package is not installed. This is similar to the
+on-demand installation of the NFS client packages in
+[Cockpit 169](https://cockpit-project.org/blog/cockpit-169.html).
+
+![Install PCP on demand](/images/pcp-install-on-demand.png)
+
+### Software Updates: Improve layout in mobile mode
+
+The Software Updates page improves spacing and layout in small browser windows
+and mobile browsers:
+
+![PackageKit mobile mode](/images/packagekit-mobile-optimization.png)
+
+
+### Remove ability to drop privileges from navigation bar
+
+Before this release, Cockpit showed a “Locked” or “Unlocked” status in the
+navigation bar. It reflected the “Reuse my password for privileged tasks”
+checkbox on the login page. Clicking on “Unlocked” would lock the interface and
+drop administrator capabilities.
+
+In general, the capability downgrade feature is not well supported across
+Cockpit. Most pages do not respond well to privilege changes.
+
+A non-clickable administrative privilege badge (“Privileged”) replaces the old
+interactive locked/unlocked status.
+
+The ability to start cockpit without escalating privileges remains on the login
+screen. Dropping privileges at runtime is still available in the user menu,
+under "Authentication".
+
+![Privilege status](/images/cockpit-drop-privileges.png)
+
+### Introduce flow control for all channels
+
+Third-party Cockpit extensions can use the Cockpit API to transfer large
+amounts of data, such as downloading an Operating System image from a remote
+machine. This now ensures that links in the route from the source to the target
+machine don't have to buffer a large, and potentially unbounded, amount of data
+when the subsequent link was slower than the previous one. This could
+previously result in the user interface getting stuck behind large data
+transfer chunks, and cockpit-bridge using an inordinate amount of memory.
+
+This release introduces a flow control mechanism to avoid these problems.
```

Comment:
The Cockpit API now has flow control to reduce buffering, improve memory usage, and make the user interface more responsive.

Third-party Cockpit extensions may use the API to transfer large amounts of data.

A notable example: [Welder](https://github.com/weldr/welder-web) downloads customized operating system from remote machines. Without flow control, Welder would become unresponsive and use large amounts of memory. This was especially noticeable with slow connections or unspecified amounts of data.